src/components/ContextProvider.tsx

```diff
@@ -25,13 +25,13 @@ export interface ContextProviderProps extends React.Props<ContextProvider> {
 export default class ContextProvider extends React.Component<
   ContextProviderProps
 > {
-  store: Store<State>;
+  store: Store<RootState>;
   admin: Admin;
   pathFor: PathFor;
 
   constructor(props) {
     super(props);
-    this.store = buildStore();
+    this.store = props.store ?? buildStore();
```

Comment:
Key change: Allow our `ContextProvider` to accept an injected store, so that it can have the same one as the Redux `Provider` component. This makes is possible to incrementally transition to the store from the `Provider` without breaking things.